icefoganalytics/travel-authorization
# Cleanup: Wire PreApproved Travel to Travel Details

Relates to:

- https://github.com/icefoganalytics/travel-authorization/issues/169

# Context

In the Approval Region of the travel details tab (https://travel-auth-dev.ynet.gov.yk.ca/my-travel-requests/78/details) can you wire the Pre-approved Travel request drop down to show Approved preapproved travel for a user based on department.

![image](https://github.com/icefoganalytics/travel-authorization/assets/89539008/264d4513-0afc-477f-a7b3-2655e4afe24d)

# Implementation

Normalize `preapproved` table to `travel_authorization_pre_approvals`.
Normalize a bunch of seeding logic.
Make a UI component for selecting travel authorization pre-approvals.

# Screenshots

N/A same as previous UI.

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. Select "PreApproved" from the top nav drop down.
5. Add a new travel pre-approval.
6. Check that all options render correctly.
7. After adding the request, select it from the table, and "submit" the selected travel.
8. Go to the "Submissioins" tab and "approve" the selected travel.
